app/controllers/settings/two_factor_authentication/otp_authentication_controller.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module Settings
+  module TwoFactorAuthentication
+    class OtpAuthenticationController < BaseController
+      include ChallengableConcern
+
+      layout 'admin'
+
+      before_action :authenticate_user!
+      before_action :verify_otp_not_enabled, only: [:show]
+      before_action :require_challenge!, only: [:create]
+
+      skip_before_action :require_functional!
+
+      def show
+        @confirmation = Form::TwoFactorConfirmation.new
+      end
+
+      def create
+        session[:new_otp_secret] = User.generate_otp_secret(32)
```

Comment:
Why is this separated like this?

Reply:
I don't know exactly why it was implemented like this, we just renamed this controller from `TwoFactorAuthenticationController` to `TwoFactorAuthentication::OtpAuthenticationController`.